common/network-common/src/main/java/org/apache/spark/network/protocol/MessageDecoder.java

```diff
@@ -40,21 +40,20 @@
   private MessageDecoder() {}
 
   @Override
-  public void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) {
-    Message.Type msgType = Message.Type.decode(in);
-    Message decoded = decode(msgType, in);
-    assert decoded.type() == msgType;
-    logger.trace("Received message {}: {}", msgType, decoded);
+  public void decode(ChannelHandlerContext ctx, ParsedFrame in, List<Object> out) {
+    Message decoded = decode(in.messageType, in.byteBuf, in.remainingFrameSize);
+    assert decoded.type() == in.messageType;
+    logger.trace("Received message {}: {}", in.messageType, decoded);
     out.add(decoded);
   }
 
-  private Message decode(Message.Type msgType, ByteBuf in) {
+  private Message decode(Message.Type msgType, ByteBuf in, long remainingFrameSize) {
     switch (msgType) {
       case ChunkFetchRequest:
         return ChunkFetchRequest.decode(in);
 
       case ChunkFetchSuccess:
-        return ChunkFetchSuccess.decode(in);
+        return ChunkFetchSuccess.decode(in, remainingFrameSize);
```

Comment:
I think you could actually decode a `StreamResponse` here with the stream Id conversion ... that might make it a lot simpler.